### PR TITLE
chore: reinstate react hooks linting, fix linting issues

### PR DESCRIPTION
**Pull request by stefl**
## Description

- Reinstates the rules of hooks eslint config
- Adds some documentation to the Readme about standards / linting / quality
- Fixes some small linting issues

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "bradlc.vscode-tailwindcss",
     "Prisma.prisma",
     "orta.vscode-jest",
-    "ms-playwright.playwright"
+    "ms-playwright.playwright",
+    "sonarsource.sonarlint-vscode"
```
Comment (stefl) [inline .vscode/extensions.json:9]: Recommend that we use the sonarsource plugin for development

File: README.md [hunk 2]
```diff
@@ -26,6 +26,16 @@ Oak AI Lesson Assistant is a project focused on experimenting with AI models and
     - [End-to-end tests](#end-to-end-tests)
       - [Playwright tags](#playwright-tags)
     - [Testing in VSCode](#testing-in-vscode)
+  - [Standards](#standards)
```
Comment (stefl) [inline README.md:29]: Add some new Readme sections

File: packages/eslint-config/src/index.mjs [hunk 3]
```diff
@@ -93,7 +93,9 @@ const config = [
 
     rules: {
       ...rules,
-      "react/jsx-no-comment-textnodes": "warn"
+      "react/jsx-no-comment-textnodes": "warn",
+      "react-hooks/exhaustive-deps": "error",
```
Comment (stefl) [inline packages/eslint-config/src/index.mjs:97]: Reinstate the hooks checks

File: packages/eslint-config/src/rules.mjs [hunk 4]
```diff
@@ -46,7 +46,7 @@ export const rules = {
   }],
 
   // Import plugin rules
-  "import/no-cycle": "warn",
+  "import/no-cycle": ["warn", { maxDepth: Infinity }],
```
Comment (stefl) [inline packages/eslint-config/src/rules.mjs:49]: Specify the circular dependency depth

File: packages/eslint-config/tsconfig.json [hunk 5]
```diff
@@ -7,8 +7,16 @@
     "outDir": "./dist",
     "declaration": true,
     "esModuleInterop": true,
-    "strict": true
+    "strict": true,
+    "baseUrl": "./src"
```
Comment (stefl) [inline packages/eslint-config/tsconfig.json:11]: Add baseUrl for imports

**Review by JBR90 — APPROVED** [review]

**Pull request merged**